DungeonMaster.cpp

```diff
@@ -0,0 +1,60 @@
+#include "DungeonMaster.h"
+
+void DungeonMaster::Fight(Unit *atk, Unit *def)
+{
+    std::cout << units[0]->ToString();
+    std::cout << units[1]->ToString();
+    std::cout << atk->GetName() << " -> " << def->GetName() << std::endl;
+
+    def->Defend(*atk);
+
+    if (def->IsDead())
+    {
+        std::cout << units[0]->ToString();
+        std::cout << units[1]->ToString();
+        std::cout << def->GetName() << " died. " << atk->GetName() << " wins." << std::endl;
+    }
+    else
+    {
+
+        Fight(def, atk);
+    }
+}
+
+DungeonMaster::DungeonMaster()
+{
+}
+
+DungeonMaster::~DungeonMaster()
+{
+    for (int i = 0; i < units.size(); i++)
+    {
+        delete units[i];
+    }
+    units.clear();
+}
+
+DungeonMaster &DungeonMaster::GetInstance()
+{
+    static DungeonMaster instance;
+    return instance;
+}
+
+void DungeonMaster::Init(char *argv[])
+{
+
+    for (int i = 0; i < 2; i++)
+    {
+        std::string name = argv[i * 3 + 1];
+
+        std::string sHp = argv[i * 3 + 2];
+        int hp = std::stoi(sHp);
+
+        std::string sDmg = argv[i * 3 + 3];
+        int dmg = std::stoi(sDmg);
+
+        units.push_back(new Unit(name, hp, dmg));
+    }
```

Comment:
ezt megcsinalhattad volna 2 sorban ugy is, hogy az ember ertse mi tortenik, es nincs  i*3+k magic, amikor i az fixen a 0-t meg 1-et veszi fel.